File: frontend/src/types/about.ts
export interface Slide {
  id: number;
  text: string;

}

export interface Story {
  id: number;
  title: string;
  slides: Slide[];
}

export interface About {
  id: number;
  top_header: string;
  top_description: string;
  bottom_header: string;
  bottom_description: string;
  img: string;
  stories: Story[];
}

File: app/schemas/about.py
from typing import List
from pydantic import BaseModel


class Slide(BaseModel):
    id:   int
    text: str
    page: int

    class Config:
        from_attributes = True


class Story(BaseModel):
    id:    int
    title: str
    order: int
    slides: List[Slide]

    class Config:
        from_attributes = True


class About(BaseModel):
    id: int
    top_header: str
    top_description: str
    bottom_header: str
    bottom_description: str
    img: str
    stories: List[Story]

    class Config:
        from_attributes = True
